rust-wasm-id-allocator/distributed-id-allocator/src/compressor/tables/uuid_space.rs

```diff
@@ -59,6 +59,38 @@ impl UuidSpace {
         }
     }
 
+    pub fn range_can_collide(
+        &self,
+        originator: SessionId,
+        sessions: &Sessions,
+        range_base: StableId,
+        range_max: StableId,
+    ) -> bool {
+        let mut range = self
+            .uuid_to_cluster
+            .range((
+                Bound::Excluded(StableId::null()),
```

Comment:
why pass null here instead of range_base? this will give incorrect results if there are multiple clusters distant from the range but sorting before range_max

Reply:
Passing range_base misses a cluster that would contain range_base. The StableId will be less than range_base.

Passing null and then reversing the iterator gets us the same behavior as with search(). We get to look at the cluster whose base is nearest to range_max. Then we just have to check whether our range_base is greater than the max Uuid in that cluster. If not, we collide.